modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotRestoreProcess.java

```diff
@@ -1134,6 +1143,31 @@ private IgniteInternalFuture<Boolean> preload(UUID reqId) {
         return retFut;
     }
 
+    /** Restore registered mappings for user classes. */
+    private void restoreMappings(File marshallerDir, BooleanSupplier stopChecker) throws IgniteCheckedException {
+        File[] mappings = marshallerDir.listFiles(BinaryUtils::notTmpFile);
+
+        if (mappings == null)
+            throw new IgniteException("Failed to list marshaller directory: " + marshallerDir);
```

Comment:
To keep some consistency across all exception messages we can change this particular message to something like that:

`Failed to list marshaller directory content [dirName=" + marshallerDir + ']'`

Actually that is up to you.